### PR TITLE
feat: add support for `XChainClaim` tx

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

This PR adds support for the `XChainClaim` transaction type.

It also replaces the `` in the tests with the actual XRP symbol, `\uE900`.

### Context of Change

cross-chain bridge tx support

### Type of Change

- [x] New feature (non-breaking change which adds functionality)

### TypeScript/Hooks Update

New files use TypeScript and Hooks.

## Test Plan

Added a basic test.

### CODE REVIEW

File: src/containers/shared/css/global.scss [hunk 1]
```diff
@@ -285,7 +285,18 @@ div.react-stockchart div {
       background: $nft;
     }
   }
+
+  &.XChainCreateBridge,
```
Comment (ckniffen) [inline src/containers/shared/css/global.scss:289]: You can use the mixin `@transaction-type-base($sidechain)`.

See https://github.com/ripple/explorer/pull/225/files#diff-cf9bdf5a51971a13a2bed1f57e95ab01982c2ef37ad6e03553275374ec945bf4R96-R101

File: src/containers/shared/components/Transaction/XChainCreateBridge/Simple.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      <SimpleRow label={t('locking_chain_issue')}>{lockingIssue}</SimpleRow>
+      <SimpleRow label={t('issuing_chain_door')}>
+        <Account account={issuingDoor} />
+      </SimpleRow>
+      <SimpleRow label={t('issuing_chain_issue')}>{issuingIssue}</SimpleRow>
+      <SimpleRow label={t('signature_reward')}>
+        <Amount value={signatureReward} />
+      </SimpleRow>
+      <SimpleRow label={t('min_account_create_amount')}>
+        <Amount value={minAccountCreateAmount ?? 'N/A'} />
```
Comment (ckniffen) [inline src/containers/shared/components/Transaction/XChainCreateBridge/Simple.tsx:39]: Is this parsed correctly.  Looking at the Amount code i will pass NaN to localizeAmount.  It has this code in it

```
if (Number.isNaN(number)) {
    return null
}
```

File: src/containers/shared/css/global.scss (hunk 1 above)
Comment (mvadari) [reply src/containers/shared/css/global.scss:289]: I haven't rebased this in a while, it's pretty outdated. I'll open it when I'm ready.

File: src/containers/Transactions/test/Description.test.js [hunk 3]
```diff
@@ -52,7 +52,7 @@ describe('Description container', () => {
...
+      '<div class="section"><div class="title">description</div><div>transaction_sequence<b> <span>31030</span></b></div>The payment is from<a class="account" title="rNQEMJA4PsoSrZRn9J6RajAYhcDzzhf8ok" href="/accounts/rNQEMJA4PsoSrZRn9J6RajAYhcDzzhf8ok">rNQEMJA4PsoSrZRn9J6RajAYhcDzzhf8ok</a>to<a class="account" title="rHoPwMC75KVUhBMeV3uDMybKG5JND74teh" href="/accounts/rHoPwMC75KVUhBMeV3uDMybKG5JND74teh">rHoPwMC75KVUhBMeV3uDMybKG5JND74teh</a><div>the_source_tag_is<b> SourceTag</b></div><div>the_destination_tag_is<b> DestinationTag</b></div><div>payment_desc_line_4<b><span> \uE9002,421.8268</span><small>XRP</small></b></div><div>payment_desc_line_6<b><span> \uE9002,421.8268</span><small>XRP</small></b></div></div>',
```
Comment (ckniffen) [inline src/containers/Transactions/test/Description.test.js:55]: Did these suddenly break?  This should be its own PR.

Comment (mvadari) [reply src/containers/Transactions/test/Description.test.js:55]: It's just editing the box to the actual XRP code

Comment (ckniffen) [reply src/containers/Transactions/test/Description.test.js:55]: Yeah i get that.  It is just tangentially related to the rest of the PR.

**Review by ckniffen — APPROVED** [review]

File: src/containers/Transactions/test/Description.test.js (hunk 3 above)
Comment (mvadari) [reply src/containers/Transactions/test/Description.test.js:55]: Yeah, that's fair. If you want me to move it to a separate PR I can do that, it just seemed like a pretty small change.

File: src/containers/shared/components/Transaction/XChainClaim/parser.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+export function parser(tx: any, meta: any) {
```
Comment (JST5000) [inline src/containers/shared/components/Transaction/XChainClaim/parser.ts:1]: Could you add a comment explaining the pre-conditions for using this function? (e.g. What type of transaction meta-data is required here?) - I'm having a bit of a hard time guessing how I could use this function correctly.

File: src/containers/shared/components/Transaction/XChainClaim/parser.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  const affectedNodes = meta.AffectedNodes
+  const modifiedAccountRoots = affectedNodes.filter(
+    (node: any) =>
+      node.ModifiedNode && node.ModifiedNode.LedgerEntryType === 'AccountRoot',
+  )
+  const doorNode = modifiedAccountRoots.filter(
+    (node: any) =>
+      node.ModifiedNode.FinalFields.Balance <
+      node.ModifiedNode.PreviousFields.Balance,
+  )[0]
```
Comment (JST5000) [inline src/containers/shared/components/Transaction/XChainClaim/parser.ts:11]: Are there any accounts that could have lost money in a transaction and NOT be the bridge owner? (Maybe this will be answered with the pre-conditions above)

File: src/containers/shared/components/Transaction/XChainClaim/test/XChainClaimSimple.test.tsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    </I18nextProvider>,
+  )
+}
+
+function expectText(
+  wrapper: ReactWrapper<any, Readonly<{}>>,
+  dataTest: string,
+  text: string,
+) {
+  expect(wrapper.find(`[data-test="${dataTest}"] .value`)).toHaveText(text)
```
Comment (JST5000) [inline src/containers/shared/components/Transaction/XChainClaim/test/XChainClaimSimple.test.tsx:26]: Why is there a space between `[data-test="${dataTest}"]` and `.value`?

File: src/containers/shared/components/Transaction/XChainClaim/parser.ts (hunk 4 above)
Comment (mvadari) [reply src/containers/shared/components/Transaction/XChainClaim/parser.ts:1]: cc @ckniffen - the new transaction format needs a bunch more documentation, but that's out of scope of this ticket. It takes the transaction shape and the metadata shape as displayed in the `tx` output.

File: src/containers/shared/components/Transaction/XChainClaim/test/XChainClaimSimple.test.tsx [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    expectText(wrapper, 'locking-chain-issue', 'XRP')
+    expectText(
+      wrapper,
+      'issuing-chain-door',
+      'rHb9CJAWyB4rj91VRWn96DkukG4bwdtyTh',
+    )
+    expect(wrapper.find(`[data-test="issuing-chain-door"] a`)).toExist()
+    expectText(wrapper, 'issuing-chain-issue', 'XRP')
+
+    expectText(wrapper, 'amount', '\uE90010.00 XRP')
```
Comment (JST5000) [inline src/containers/shared/components/Transaction/XChainClaim/test/XChainClaimSimple.test.tsx:49]: Why do we use the character `\uE900`?

File: src/containers/shared/components/Transaction/XChainClaim/test/XChainClaimSimple.test.tsx (hunk 6 above)
Comment (mvadari) [reply src/containers/shared/components/Transaction/XChainClaim/test/XChainClaimSimple.test.tsx:26]: That's just how the `wrapper.find` method works - maybe @ckniffen has more details on that

File: src/containers/shared/components/Transaction/XChainClaim/test/XChainClaimSimple.test.tsx (hunk 7 above)
Comment (mvadari) [reply src/containers/shared/components/Transaction/XChainClaim/test/XChainClaimSimple.test.tsx:49]: That's unicode for the XRP symbol

File: src/containers/shared/components/Transaction/XChainClaim/test/XChainClaimSimple.test.tsx (hunk 6 above)
Comment (ckniffen) [reply src/containers/shared/components/Transaction/XChainClaim/test/XChainClaimSimple.test.tsx:26]: It is a css selector which means find the element(s) with the  attribute `data-test` set to something and then find the element(s) with the class name of `value`

File: src/containers/shared/components/Transaction/XChainClaim/test/XChainClaimSimple.test.tsx (hunk 7 above)
Comment (ckniffen) [reply src/containers/shared/components/Transaction/XChainClaim/test/XChainClaimSimple.test.tsx:49]: That is the character code used for XRP symbol

File: src/containers/shared/components/Transaction/XChainClaim/parser.ts (hunk 5 above)
Comment (mvadari) [reply src/containers/shared/components/Transaction/XChainClaim/parser.ts:11]: Hmm my example is submitted by the door account (aka the bridge owner). Normally, two accounts will lose money: the account that pays the fee and the door account. So I'll have to think about that a little more.

File: src/containers/shared/components/Transaction/XChainClaim/parser.ts (hunk 4 above)
Comment (ckniffen) [reply src/containers/shared/components/Transaction/XChainClaim/parser.ts:1]: @JST5000 @ckniffen Does this clarify it better?

Comment (JST5000) [reply src/containers/shared/components/Transaction/XChainClaim/parser.ts:1]: I think I'm clear on what the parser does, it just seemed like this function wouldn't produce the correct results if you passed in any transaction. And since it required a specific kind of transaction in order to produce the right results, I was thinking that'd be good to include as a comment here. (Otherwise I may accidentally treat it's type signature as the only limitation on it's use)

Comment (mvadari) [reply src/containers/shared/components/Transaction/XChainClaim/parser.ts:1]: Yeah there are still some refactors that need to be made to the transaction system to really make it easy to read/use. But the current system is so much better than how it used to be.

File: src/containers/shared/components/Transaction/XChainClaim/parser.ts (hunk 5 above)
Comment (JST5000) [reply src/containers/shared/components/Transaction/XChainClaim/parser.ts:11]: Is this resolved now, or not something which needs to be resolved in this PR?

**Review by JST5000 — APPROVED** [review]
LGTM
